packages/ui/src/utils/form.ts

```diff
@@ -11,3 +26,37 @@ export const passwordErrorWatcher = (error?: FieldError): ErrorType | undefined
     default:
   }
 };
+
+export const identifierErrorWatcher = (
+  enabledFields: IdentifierInputType[],
+  error?: FieldError
+): ErrorType | undefined => {
+  const data = { types: enabledFields.map((field) => t(identifierInputPlaceholderMap[field])) };
+
+  switch (error?.type) {
+    case 'required':
+      return {
+        code: 'general_required',
+        data,
+      };
+    case 'validate':
+      return {
+        code: 'general_invalid',
+        data,
+      };
+    default:
+  }
+};
```

Comment:
Would be better to use a hook for better performance. e.g. useMemo

Reply:
will be removed in the following PR.